src/main/java/site/coach_coach/coach_coach_server/coach/service/CoachService.java

```diff
@@ -159,20 +159,22 @@ public Coach getCoachById(Long coachId) {
 			.orElseThrow(() -> new NotFoundException(ErrorMessage.NOT_FOUND_COACH));
 	}
 
-	@Transactional(readOnly = true)
-	public Coach getCoachByUserId(User user) {
-		return coachRepository.findByUser_UserId(user.getUserId())
-			.orElseThrow(AccessDeniedException::new);
-	}
-
 	@Transactional(readOnly = true)
 	public CoachDetailDto getCoachDetail(User user, Long coachId) {
-		Coach coach = (coachId != null) ? getCoachById(coachId) : getCoachByUserId(user);
+		Coach coach = (coachId != null) ? getCoachById(coachId) : getCoachByUserId(user.getUserId());
+
+		boolean isSelf = user.getUserId().equals(coach.getUser().getUserId());
+
+		if (!isSelf && !coach.getIsOpen()) {
+			throw new AccessDeniedException();
+		}
 
 		List<ReviewDto> reviews = getReviews(coach);
 		double averageRating = calculateAverageRating(reviews);
 
 		boolean isLiked = isLikedByUser(user, coach);
+		boolean isContacted = matchingRepository.existsByUserUserIdAndCoachCoachId(user.getUserId(), coachId);
```

Comment:
이때 coachId가 null 일때는 필요없는 부분 맞나요???

```suggestion
		boolean isContacted = (coachId != null) && matchingRepository.existsByUserUserIdAndCoachCoachId(user.getUserId(), coachId);
```
그럼 요롷게 해주면 불필요한 로직을 피할 수 있을 것 같습니다! 사용자가 본인의 코치 정보를 볼 때에는 매칭 상태를 확인할 필요가 없다는 생각이 들어서요!!